Docs deploy: `npm ci` fails with EBADPLATFORM on transitive optional esbuild binary

## Problem

After #371 switched `Install dependencies` to `npm ci`, the docs-deploy job fails:

```
Run npm ci
npm error code EBADPLATFORM
npm error notsup Unsupported platform for @esbuild/aix-ppc64@0.28.0: wanted {"os":"aix","cpu":"ppc64"} (current: {"os":"linux","cpu":"x64"})
npm error notsup Valid os:   aix
npm error notsup Actual os:  linux
npm error notsup Valid cpu:  ppc64
npm error notsup Actual cpu: x64
Error: Process completed with exit code 1.
```

## Root cause

`vite@8.0.13` declares `esbuild` only as a `devDependency` and as an optional `peerDependency`. With no parent dependency chain pulling esbuild in, npm records `node_modules/esbuild@0.28.0` as `"extraneous": true` in `docs/package-lock.json` (along with its `optionalDependencies` listing all `@esbuild/<os>-<cpu>` binaries).

The CI runner's npm (Node 24 → npm 11.x) hits a platform-handling bug for these transitive optional binaries and refuses `@esbuild/aix-ppc64` with `EBADPLATFORM` instead of skipping it. Local npm 11.13 does not exhibit this; a fresh `rm -rf node_modules package-lock.json && npm install` reproduces the exact same lockfile bytes and `npm ci` succeeds against it.

## Proposed fix

Revert `npm ci` back to `npm install` (which is unaffected by the bug), and address the original failure (#370) directly: drop the lockfile mutation right before the `gh-pages` checkout so the working tree is clean.

`.github/workflows/docs-deploy.yml`:

```yaml
      - name: Install dependencies
        run: npm install              # back from `npm ci`
        working-directory: docs

      ...

      - name: Checkout gh-pages branch
        run: |
          # `npm install` may refresh docs/package-lock.json (integrity hashes,
          # metadata). Discard that mutation so the branch switch doesn't abort.
          git checkout -- docs/package-lock.json
          git fetch origin gh-pages:gh-pages 2>/dev/null || true
          ...
```

This sidesteps the npm version edge case entirely while still solving #370.

## Why not regenerate the lockfile

Already tried. `rm -rf node_modules package-lock.json && npm install` produces an identical lockfile, so this is not a corruption issue — it's runtime npm behavior in CI.

## Impact

Blocks docs deploy on `main` (continuation of #368, #370, originally diagnosed in the now-closed #372).

--- a/.github/workflows/docs-deploy.yml
+++ b/.github/workflows/docs-deploy.yml
@@ -104,7 +104,7 @@ jobs:
           cache-dependency-path: 'docs/package-lock.json'
 
       - name: Install dependencies
-        run: npm ci
+        run: npm install
         working-directory: docs
 
       - name: Build documentation
@@ -118,6 +118,9 @@ jobs:
 
       - name: Checkout gh-pages branch
         run: |
+          # `npm install` may refresh docs/package-lock.json (integrity hashes,
+          # metadata). Discard that mutation so the branch switch doesn't abort.
+          git checkout -- docs/package-lock.json
           git fetch origin gh-pages:gh-pages 2>/dev/null || true
           if git rev-parse --verify gh-pages >/dev/null 2>&1; then
             git checkout gh-pages

